cv_next/components/layout/navbar/popupToggle.tsx

```diff
@@ -1,10 +1,36 @@
 "use client";
 import profileIcon from "@/public/images/profile.png";
 import Image from "next/image";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import Popup from "./popup";
+import { useSupabase } from "@/hooks/supabase";
+import { getUserFromId } from "@/app/actions/user/fetchUserInfo";
 
 export function PopupToggle() {
+  const [profileImage, setProfileImage] = useState<any>(profileIcon);
+  const [userData, setUserData] = useState<UserModel | null>(null);
+  const [signedOut, setSignout] = useState(true);
```

Comment:
Semantics: it's easier to comprehend a state that controls "signIn" rather than "signOut".